bin/start_gcb_build.sh

```diff
@@ -0,0 +1,262 @@
+#!/bin/bash
+# Copyright 2017 Istio Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+################################################################################
+
+# This is an example file for how to start an istio build using Cloud Builder.
+# To run it you need a Google Cloud project and the key file for a service
+# account that had been granted access to start a build.  This example
+# assumes your GCR and GCS buckets have the same name as your project.
+#
+# An example invocation script is:
+# 
+# PROJECT_ID=${USER}-experimental
+# KEY_FILE=~/${PROJECT_ID}-cloudbuild.json
+# MFEST_URL=https://github.com/istio/green-builds
+# MFEST_FILE=build.xml
+# MFEST_COMMIT=master
+# VERSION="0.3.0"
+# ./start_build.sh -w -p "${PROJECT_ID}" -r "${PROJECT_ID}" -s "${PROJECT_ID}"
+#           -b "" -c "builds/${VERSION}" -v "${VERSION}" -k "${KEY_FILE}"
+#           -u "${MFEST_URL}" -m "${MFEST_FILE}" -t "${MFEST_COMMIT}
+
+set -o nounset
```

Comment:
you should add
set -o errexit
set -o pipefail
As stuff will fail sliently

Reply:
I tried adding it but got a reminder on why I didn't use them.  parse_result_file() returns 0 on build success, 1 if build running, and 2 if build fails.  The act of doing a non-zero return causes the script to bail (and if this didn't cause a bail then the callee would likely bail when it saw the result).  I'm not sure this script will even be used in production in the long term--it'd probably be better to use a language that can leverage the cloud APIs.

Reply:
OK please add error handling for all calls then such that we don't siliently fail.

function error_exit() {echo ${1}; exit 1; }

command || error_exit "Failed to run command"